### DISCUSSION TITLE
Using pg-boss as a Queue Handler: Prisma Compatibility ?

### DISCUSSION THREAD

**Discussion by lgauthier1** [timgit/pg-boss#391] [Q&A] [answered]
I am considering replacing Redis by PostgreSQL to handle my queue for my data integration pipeline.
I only use Redis as a queue, which complicates my architecture and D.R.P. (Disaster Recovery Plan) unnecessarily.

Therefore, I am considering replacing Redis (via the Bull package) with pg-boss.

I use Prisma as an ORM for my database. Have you ever been in this situation?

if you have any advice or feedback regarding the use of pg-boss with Prisma as an ORM for handling queues , I would be happy to hear them. Specifically, any insights on how to effectively integrate pg-boss with Prisma or any potential challenges to watch out for would be greatly appreciated. Thank you!

**Comment by lgauthier1** [comment, marked as answer]
I just ran some tests to find the answer (Find repository here: https://github.com/lgauthier1/pgboss-with-prisma), and pg-boss is compatible with Prisma. `pgboss` creates its own schema (pgboss) to manage different queues.

Therefore, there is no conflict between Prisma and your current schema and pg-boss.
<img width="340" alt="image" src="https://user-images.githubusercontent.com/37870500/236637375-7c2ac5f0-f636-405a-b1f1-beb7ddad7544.png">

**Comment by kartomic8** [comment]
FYI, it looks like you're all set, but in case it helps, I have the two frameworks integrated, and its working smoothly.  In my use case, I regularly need to send messages to queues transactionally. 

You can enlist pg-boss operations into  db transactions on the prisma client using:
1. PgBoss send options.
2. Prisma interactive transactions and the Prisma.queryRaw API.

**Reply by lgauthier1** [reply]
Thanks for your advices.
